src/app/PublicApp.js

```diff
@@ -46,6 +47,7 @@ const PublicApp = ({ onLogin }) => {
                         path="/pre-invite/:selector/:token"
                         render={({ history, match }) => <PreInviteContainer history={history} match={match} />}
                     />
+                    <Route exact path="/signup" render={() => <SignupContainer onLogin={onLogin} />} />
```

Comment:
Get `history` from `render` instead of `withRouter`.
```suggestion
                    <Route exact path="/signup" render={({ history }) => <SignupContainer history={history} onLogin={onLogin} />} />
```

Reply:
Ok, but why though?

Reply:
It's just to avoid the unnecessary wrapping of `withRouter` when it can be passed directly as props. I'd like to have hooks to avoid this kind of thing. :P